rewrite-xml/src/main/java/org/openrewrite/xml/XsltTransformation.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2020 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.xml;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+import org.intellij.lang.annotations.Language;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.HasSourcePath;
+import org.openrewrite.Option;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.Validated;
+import org.openrewrite.internal.StringUtils;
+import org.openrewrite.internal.lang.Nullable;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.Charset;
+import java.util.Arrays;
+
+@Slf4j
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class XsltTransformation extends Recipe {
+
+    @Nullable
+    @Language("xml")
+    @Option(displayName = "XSLT Configuration transformation",
+            description = "The transformation to be applied.")
+    String xslt;
+
+    @Nullable
+    @Option(displayName = "XSLT Configuration transformation classpath resource",
+            description = "Recipe transformation provided as a classpath resource.")
+    String xsltResource;
+
+    @Option(displayName = "File pattern",
+            description = "A glob expression that can be used to constrain which directories or source files should be searched. " +
+                          "Multiple patterns may be specified, separated by a semicolon `;`. " +
+                          "If multiple patterns are supplied any of the patterns matching will be interpreted as a match.",
+            example = "**/*.xml")
+    String filePattern;
```

Comment:
I'm wondering if it would make sense to add an xpath expression option to this recipe, such that we might not have to add the xslt options to `ChangePluginConfiguration`. This recipe can then match particular tags within specific files, and be used in isolation. Otherwise I might foresee XSLT options being added to more recipes, whereas that would complicate maintenance a bit going forward. Curious to hear how you'd feel about that.

Reply:
The ChangePluginConfiguration, in my use case, its' the simplest transformation. The general case XsltTransfomation is being used in my setup to convert OSGI / Karaf blueprints to Spring Boot.  The XsltTransformation can do search & replace on XML files without converting them to PlainText files as wel..